### PR TITLE
Always use the database instance declared by devs

### PR DESCRIPTION
**Pull request by fastslack**

### CODE REVIEW

File: libraries/joomla/table/menu.php [hunk 1]
```diff
@@ -110,14 +110,6 @@ public function check()
 			return false;
 		}
 
-		// Verify that a first level menu item alias is not the name of a folder.
```
Comment (elinw) [inline libraries/joomla/table/menu.php:113]: What does this have to do with the topic of the pull request?

Comment (fastslack) [inline libraries/joomla/table/menu.php:113]: You can discard this lines remove, i commit that after send this pull request but i dont know how to delete the commit :-s
The important fix is the changes on JTable.
Thanks

**Comment by elinw** [comment]
The easiest thing (from experience) is just to make another branch and do the commit again there.

**Comment by jonnsl** [comment]
@fastslack
git checkout tables
git reset --hard HEAD~1
git push origin tables --force

File: libraries/joomla/table/menu.php (hunk 1 above)
Comment (eddieajau) [inline libraries/joomla/table/menu.php:113]: I think you need to discard this branch and just start again.
